doc/design/docker-cp.md

```diff
@@ -0,0 +1,146 @@
+ 
+ Additionally, it must be ensured that regardless of error or success the disk must be detached from the VCH. We must ensure that this also occurs when we boot up the VCH in the event that the disk is attached to the VCH and is hard shutdown. This code already exists and should automatically take effect for us. The merrits for calling `fsck` have not been determined yet. Investigation will be needed to see what kind of writes and operations could make us susceptible to needing `fsck`. 
+ 
+ 
+ Docker cp behavior is as such(below), however the portlayer behavior is not dictated by the personality behavior so we must find a way to reconcile how we allow users to set permissions. Likely we should have the personality specify the UID/GID which it would like the portlayer to assign to the imported archive. 
+ 
+ 
+``` files copied to a container are created with UID:GID of the root user. Files copied to the local machine are created with the UID:GID of the user which invoked the docker cp command.```
+ 
+ In An Ideal scenario without any failures the process will be as such : 
+ 
+ ```
+ 1) get request from personality for a file copy, it will have the stream to readfrom/write to as well as a path and a container ID.
+ 
+ 2) depending on the endpoint hit depends on the to/from scenario either way in an *off* scenario we would first get the container object and confirm that it is off. 
+ 
+ 3) Once confirmed that the container is off. We will make the request to mount for the import/export operation. 
+ 
+ 4) Then we will mount the r/w layer at a temp directory location and block all start operations on this container as we have the r/w already mounted to different compute device. In the even of a `copyFrom` scenario the disk should be mounted as `non-persistent`
+ 
+ 5) Begin streaming the tar file to the mounted disk at the assembled path. 
+ 
+ 6) We will need to untar this in place while checking for existing files and behaving according to the overwrite dir flag. We will need this to be in place since we could be operating with limited space, we want to do this in a fashion that we have as little storage overhead during unpacking as possible. Checking flags for `untar` may help significantly with this. Otherwise we may need to read the tar headers themself before attempting the write and determining if we have the size to do this.  
+ 
+ 
+ 
+ 7) Upon successfully copying the data we will adjust the permissions to be appropriately aligned with our decisions. Then detach the disk.
```

Comment:
I have looked at @dougm stuff, We likely do not need to worry about it. I would like to leave it there so that readers understand that we do have an obligation to change the permissions after the untar.